Jenkinsfile

```diff
@@ -1098,7 +1117,6 @@ pipeline {
                         'Functional Hardware Medium': getFunctionalTestStage(
                             name: 'Functional Hardware Medium',
                             pragma_suffix: '-hw-medium',
-                            base_branch: 'master',
```

Comment:
Why is this not needed anymore?

Reply:
It only should be specified for daily/weekly branches to specify what branch to use.

If you specify it on master or release branches it should be a noop.

If you specify it on a PR, then it is not running test code in the PR as intended and you end up landing test code that has never been actually tested in a PR.

Reply:
That parameter swaps out everything checked out for the PR except the Jenkinsfile for the code in the master branch.

Any changes to any files in the repository that affect testing in the PR will actually just be using the master branch instead of the new code.

This was exposed in this PR because we have bugfixes in this PR to prevent CI from failing, and until this change is landed, all PRs being run are not actually running the test code that the PR author intended to be run.

Reply:
I don't doubt there is an issue here but
> Any changes to any files in the repository that affect testing in the PR

I don't think is correct. Otherwise we wouldn't be able to validate test fixes in PRs, which we do all the time. I can modify some files and my changes are in fact picked up in the functional test stages.

Reply:
Some test code is compiled and stored in DAOS RPMs, that code gets tests.
None of the code for doing CI provisioning that may have changed in the PR will be used in the PR.